Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebContext.cpp

```diff
@@ -1043,6 +1044,31 @@ static void testWebContextTimeZoneOverrideInWorker(WebViewTest* test, gconstpoin
     g_assert_cmpstr(WebViewTest::javascriptResultToCString(javascriptResult), ==, "Europe/Berlin, Europe/Berlin, Europe/Berlin, Europe/Berlin");
 }
 
+static void testNoWebProcessLeakAfterWebKitWebContextDestroy(WebViewTest* test, gconstpointer)
+{
+    webkitSetCachedProcessSuspensionDelayForTesting(0);
+    GRefPtr<WebKitWebContext> webContext = adoptGRef(WEBKIT_WEB_CONTEXT(g_object_new(WEBKIT_TYPE_WEB_CONTEXT, nullptr)));
+    GRefPtr<WebKitWebView> webView = Test::adoptView(Test::createWebView(webContext.get()));
+    webkit_web_view_load_uri(webView.get(), kServer->getURIForPath("/").data());
+    test->waitUntilLoadFinished(webView.get());
+    bool didRunForceRepaingCallback = false;
+    webkitWebViewForceRepaintForTesting(webView.get(), [] (void* data) {
+        *static_cast<bool*>(data) = true;
+    }, &didRunForceRepaingCallback);
+    webView.clear();
+    // Wait for shutdownPreventingScope created in WebPageProxy::close to be destroyed.
+    test->wait(0.5);
```

Comment:
Meh, I don't like the hardcoded timeouts in the test, but I suppose it's better than having no test.

Bonus points for writing this test in the first place. I would have considered this situation hard to test, and not expected you to attempt it, but you even added new internal APIs to make it possible. Good.

Reply:
> I don't like the timeout either but I don't see a better way to ensure that the task scheduled in the ~WebPageProxy on the main loop actually gets dispatched. If you have a better way than the timeout in mind I can update it.

Is it sufficient to iterate the default main context? `g_main_context_iteration (NULL, TRUE)`?

Reply:
Yes, that works. Updated the test.